### PR TITLE
test: expand mock tests for 28 executor handler files

### PR DESCRIPTION
**Pull request by retran**
## Why

The mxcli project is preparing to extract the `mdl/` runtime into a reusable library. Before any refactoring can safely begin, comprehensive test coverage is needed to catch behavioral regressions. PR #253 (C1) added initial mock tests for 8 handler files that had zero coverage. This PR completes the executor handler coverage by expanding the remaining 28 handler files that had only minimal tests (typically just a happy-path case).

Without these tests, error paths, filter logic, and edge cases in these handlers would go unvalidated during refactoring — exactly the kind of subtle breakage that is hardest to catch.

Part of the stacked PR series: #253 (C1) → #254 (C2) → #255 (D1).

## Summary

- Expand mock test coverage across 28 existing `*_mock_test.go` files that previously had minimal tests
- Adds ~95 new tests covering: not-found errors, filter-by-module, backend error propagation, JSON format output, edge cases
- Builds on PR #253 (C1) which added the initial 8 handler test files

## Batch breakdown

| Commit | Files | Tests added | Handlers |
|--------|-------|-------------|----------|
| 1 | 4 | 22 | settings, fragments, misc, javaactions |
| 2 | 6 | 8 | modules, entities, associations, enumerations, constants, microflows |
| 3 | 6 | 14 | odata, published_rest, rest_clients, businessevents, workflows, javascript_actions |
| 4 | 8 | 23 | datatransformer, dbconnection, imagecollections, jsonstructures, import/export_mappings, mermaid, navigation |
| 5 | 4 | 36 | pages, security, write_handlers, agenteditor |

## What's covered

- **Not-found paths** for all describe/drop handlers
- **Filter-by-module** for list handlers that support it
- **Backend error propagation** for handlers not covered by cross-cutting error tests
- **JSON format output** with `assertValidJSON` for list handlers
- **Edge cases**: nil-name validation, unsupported type errors, already-exists guards on create handlers, empty result sets

## Notes

- Cross-cutting test files (`cmd_notconnected_mock_test.go`, `cmd_error_mock_test.go`, `cmd_json_mock_test.go`) already cover not-connected, backend errors, and JSON for many handlers — new tests avoid duplication
- Write handler create/alter tests (OData, REST, business events) deferred — complex multi-backend setup
- 28 files changed, ~1480 insertions, 0 deletions

### CODE REVIEW

File: mdl/executor/cmd_lint_mock_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	err := execLint(ctx, &ast.LintStmt{})
+	assertError(t, err)
+	assertContainsStr(t, err.Error(), "not connected")
+}
+
+// ---------------------------------------------------------------------------
+// listLintRules — ShowRules happy path
+//
+// The ShowRules=true path calls listLintRules which only needs ctx.Output.
+// It prints built-in rules (ID + Name + Description + Category + Severity).
```
[the remark this answers is not in the document]
Comment (retran) [reply mdl/executor/cmd_lint_mock_test.go]: Good catch — updated the comment to clarify that `execLint` checks `Connected()` before reaching the ShowRules branch. Fixed in dccb33d.

File: mdl/executor/cmd_move_mock_test.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package executor
+
+import (
```
[the remark this answers is not in the document]
Comment (retran) [reply mdl/executor/cmd_move_mock_test.go:5]: This file (`cmd_move_mock_test.go`) was introduced in PR #253 (C1), not in this PR. C2 is stacked on C1, so the diff against `main` includes C1's files — but C2 itself only expands existing test files. The PR description is accurate for C2's scope.

**Comment by ako** [comment]
## Code Review

### Overview

This is C2 in the stacked series (#253 → #254 → #255), adding ~95 tests across 28 executor handler files. The coverage is systematic and well-organized: not-found paths, filter-by-module, backend error propagation, JSON output, and edge cases. The test structure is consistent throughout and the inline documentation comments (explaining WHY certain paths are left uncovered, with \"tracked separately\" notes) are a good pattern.

### Blocking issue

**`TestHelp_Mock` will not compile.**

The test calls `execHelp(ctx)` with one argument, but the current `cmd_misc.go:55` signature is:

```go
func execHelp(ctx *ExecContext, s *ast.HelpStmt) error {
```

This was changed during the PR #172 rebase (syntax feature registry). The fix is either:

```go
// Option A — test the no-topic path
assertNoError(t, execHelp(ctx, &ast.HelpStmt{}))
```

or extract a zero-arg wrapper if the REPL really does call `execHelp(ctx)` directly (it doesn't — `register_stubs.go` passes `stmt.(*ast.HelpStmt)`). Option A is correct.

### Non-blocking issues

1. **`cmd_misc_mock_test.go` — `TestHelp_Mock`**: See above. Also check that `execHelp(ctx, &ast.HelpStmt{})` actually emits "MDL Commands" / "connect local" in the no-topic branch (the assertions are reasonable, just verify they match the actual output format).

2. **`cmd_mermaid_mock_test.go` — `TestDescribeMermaid_UnsupportedType`**: Uses `fmt.Sprint(err)` instead of `err.Error()`. All other error string assertions in this PR use `err.Error()` — this one is an inconsistency, though not a bug.

3. **`cmd_features_mock_test.go` — `TestShowFeatures_InArea_ForVersion`**: The assertion `assertContainsStr(t, buf.String(), "domain_model")` relies on the area filter header format emitting the literal string `domain_model`. If the output says "domain model" (with a space) or uses a display name, this test will fail spuriously. Worth verifying against the actual `execShowFeatures` output format.

4. **`cmd_lint_mock_test.go` — comment accuracy**: The comment says "execLint currently checks ctx.Connected() before dispatching to the ShowRules branch, so this test still needs a connected backend." If that's true, it means `SHOW LINT RULES` fails when disconnected — which seems like a UX bug in the implementation, not in the test. Worth flagging as a separate issue if the ShowRules path doesn't actually need a live project.

### Strengths

- **`TestDropFolder_NestedPath`**: verifies that the *child* folder ID (not the parent) is passed to `DeleteFolder` — exactly the kind of off-by-one that integration tests miss.
- **`TestMoveFolder_ToModule` / `TestMoveFolder_ToFolder`**: captures the `movedTo` ID and asserts it matches the destination — good mutation verification pattern.
- **`TestAlterPage_*` suite**: thorough coverage of the mutator lifecycle (open, operate, save, and each error path).
- **Cross-reference comments** in `cmd_constants_mock_test.go`, `cmd_enumerations_mock_test.go`, `cmd_microflows_mock_test.go`: helpful pointers to where cross-cutting tests live, without duplicating them.
- **`cmd_import_mock_test.go` / `cmd_lint_mock_test.go`**: honest about coverage limitations with "tracked separately" notes rather than writing shallow tests that give false confidence.

### Summary

One hard compile error (`TestHelp_Mock` argument count mismatch) needs to be fixed before this can merge. Everything else is either a minor style inconsistency or a note for follow-up. The test structure and scope are solid.
